### ISSUE TITLE
Allow datasource setup without interfering defaults 

### ISSUE THREAD

**Issue by remmeier** [corda/corda#4641]
Current the reference.conf provides a default for the datasource:

```
dataSourceClassName = org.h2.jdbcx.JdbcDataSource
dataSource.url = "jdbc:h2:file:"${baseDirectory}"/persistence;DB_CLOSE_ON_EXIT=FALSE;WRITE_DELAY=0;LOCK_TIMEOUT=10000"
dataSource.user = sa
dataSource.password = ""
```

while it can be convenience for a basic test/dev setups, it is not useful for production setup (as nobody will make use of H2) and in worst case can interfere and cause issues with the production setup. Some of our nodes we deploy to Google Cloud, for example. There we make use of managed SQL and use a special SockerFactory (cloud sql socket proxy) that handles the connectivity/authentication of Corda to the the database. It is a bit of a maze in terms of parameters. Among others, Hikari as two modes of establishing connections, through datasource.url and jdbcurl. The setup we have as far as I (and others) can tell only works with the later, so our configuration rather looks like:
```
dataSourceProperties = {
    # we use jdbcUrl rather than url for Driver-based setup, see https://github.com/brettwooldridge/HikariCP#configuration-knobs-baby
    jdbcUrl = ...
    username = ....
    password = ...
}
```

as the generic Postgresql driver is unable to pass the google-specifc  socketFactory properties.

Long story short, in such a setup the defaults interfere and both jdbcUrl and datasource.url get configured, which does not work. It should be possible to be in full control of the Corda datasource properties. So the defaults should either:

- be removed
- check for the presence of datasource.url and jdbcurl
- be disabled if any datasource-related properties are configured.

in terms of compatiblity with current behavior, maybe the second could be the way to go. we currently have our own corda branch and did go with the first solution.

**Comment by nargas-ritu** [comment]
Internal ticket - [CORDA-4246](https://r3-cev.atlassian.net/browse/CORDA-4246)

**Comment by peterli-r3** [comment]
Hi remmeier,

With Corda 5 coming soon, we are closing up the issues prior to Corda v4.5. Thank you for your contribution to Corda!

Please feel free to contact [devrel@r3.com](mailto:devrel@r3.com) if you need any further information
